content_scripts/storageLoader.js

```diff
@@ -219,6 +219,46 @@ function listenForBackgroundMessages() {
     });
 }
 
+/**
+ * General method to scrape DOM inside html
+ * @param url   The url consisting the DOM element
+ * @param cssSelector   CSS selector to find the DOM element
+ * @param all   Get all DOM elements matching the selector or just the first one (default false)
+ * @returns {Promise<NodeListOf<Element>|Element>}  Array of dom elements in case sent with all param "true"
+ *                                                   otherwise just one DOM element
+ */
+async function scrapeWebsiteDOM(url, cssSelector, all = false) {
+    let parser = new DOMParser()
+    let response = await fetch(url)
+    const contentType = response.headers.get('Content-Type')
+    const charsetStartIndex = contentType.lastIndexOf("charset=")
+    let htmlDoc
+    if (charsetStartIndex !== -1) {
+        const charSet = contentType.substring(charsetStartIndex + 8)
+        const html = new TextDecoder(charSet).decode(await response.arrayBuffer())
+        htmlDoc = parser.parseFromString(html, 'text/html')
+    } else {
+        htmlDoc = parser.parseFromString(await response.text(), 'text/html')
+    }
+    return all ? htmlDoc.querySelectorAll(cssSelector) : htmlDoc.querySelector(cssSelector)
+}
+
+/**
+ *  Initial scrapping functionality from Syllabus to get in which semester is a course by Syllabus content
+ * @param courseId  The ID of the course to check
+ * @returns {Promise<string>} "a" for semester A and "b" for semester B
+ */
+async function getCourseSemester(courseId) {
+    if (courseId.length < 5) {
```

Comment:
also here, why are we checking that? why number 5?

Reply:
I'll add a comment and maybe constant, but for now, I'll explain here too.
If the course id is with less than 5 chars then in the URL for the course's syllabus they add a prefix of 0s so the id will be with 5 chars.